### PR TITLE
Add docs on docker root user

### PR DESCRIPTION
**Pull request by MasterPtato**
Fixes WEB-56

### CODE REVIEW

**Comment by MasterPtato** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/site/95?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #90** <a href="https://app.graphite.dev/github/pr/rivet-gg/site/90?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #95** <a href="https://app.graphite.dev/github/pr/rivet-gg/site/95?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/site/95?utm_source=stack-comment).

File: src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Docker root user
```
Comment (MasterPtato) [inline src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx:1]: @NathanFlurry relative changes here

File: src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    FROM sickcodes/docker-osx:latest
+
+    WORKDIR /app
+    COPY . .
+
+    RUN dscl . -create /Users/myuser
+    USER myuser
+
+    CMD run.sh
+    ```
```
Comment (NathanFlurry) [inline src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx:81]: We don't need windows & macos.

File: src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Docker root user
+
+Although Docker allows running containers as the root user, Rivet does not allow uploaded build images to use
```
Comment (NathanFlurry) [inline src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx:3]: I think a lot of people will come in to this not understanding how linux users work/why they're important. This paragraph is good, but might be good to add a "learn more" link to something that describes users in dockerfiles in an easy to understand way.

Comment (NathanFlurry) [reply src/pages/docs/dynamic-servers/concepts/docker-root-user.mdx:3]: This whole document is _scary_ to someone who doesn't know how to use a cli. trying to make it less scary.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

**Comment by AngelOnFira** [comment]
@MasterPtato one thing I've found is separating PRs in the stack into one that just has formatting and one that has all the important changes. Makes it a bit easier to go through I think 🙌

**Review by AngelOnFira — APPROVED** [review]

**Comment by MasterPtato** [comment]
> @MasterPtato one thing I've found is separating PRs in the stack into one that just has formatting and one that has all the important changes. Makes it a bit easier to go through I think 🙌 

Will do next time, thanks

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
Should split up the section in to "What is a root user?" and "How do I run my Docker container as a non-root user?"

Put "Rivet does not allow uploaded build images to run as the root user" in the intro paragraph

can't review specific code because of fmt. i'm planning on running a big fmt and re-enabling ci on it once all of these chnages are merged.

**Comment by MasterPtato** [comment]
@NathanFlurry hows this? i think its more cohesive

**Comment by NathanFlurry** [comment]
> @NathanFlurry hows this? i think its more cohesive
> 
> 
> 
> 

lgtm

**Review by NathanFlurry — APPROVED** [review]
Should split up the section in to "What is a root user?" and "How do I run my Docker container as a non-root user?"

Put "Rivet does not allow uploaded build images to run as the root user" in the intro paragraph

can't review specific code because of fmt. i'm planning on running a big fmt and re-enabling ci on it once all of these chnages are merged.

**Pull request merged**